pythonforandroid/recipes/android/src/android/storage.py

```diff
@@ -0,0 +1,90 @@
+
+from jnius import autoclass, cast
+import os
+
+
+def app_storage_path():
+    """ Locate the built-in device storage used for this app only.
+
+        This storage is APP-SPECIFIC, and not visible to other apps.
+        It will be wiped when your app is uninstalled.
+
+        Returns directory path to storage.
+    """
+    PythonActivity = autoclass('org.kivy.android.PythonActivity')
+    currentActivity = cast('android.app.Activity',
+                           PythonActivity.mActivity)
+    context = cast('android.content.ContextWrapper',
+                   currentActivity.getApplicationContext())
+    file_p = cast('java.io.File', context.getFilesDir())
+    return os.path.normpath(os.path.abspath(
+        file_p.getAbsolutePath().replace("/", os.path.sep)))
+
+
+def primary_external_storage_path():
+    """ Locate the built-in device storage that user can see via file browser.
+        Often found at: /sdcard/
+
+        This is storage is SHARED, and visible to other apps and the user.
+        It will remain untouched when your app is uninstalled.
+
+        Returns directory path to storage.
+
+        WARNING: You need storage permissions to access this storage.
+    """
+    Build = autoclass('android.os.Build')
+    if Build.VERSION.SDK_INT >= 24:
+        Environment = autoclass('android.os.Environment')
+        sdpath = (
+            Environment.getExternalStorageDirectory().getAbsolutePath()
+        )
+        # Apparently this can both return primary (built-in) or
+        # secondary (removable) external storage depending on the device,
+        # therefore check that we got what we wanted:
+        if not Environment.isExternalStorageRemovable(sdpath):
+            return sdpath
+    if "EXTERNAL_STORAGE" in os.environ:
+        return os.environ["EXTERNAL_STORAGE"]
+    raise RuntimeError(
+        "unexpectedly failed to determine " +
+        "primary external storage path"
+    )
+
+
+def secondary_external_storage_path():
+    """ Locate the external SD Card storage, which may not be present.
+        Often found at: /sdcard/External_SD/
+
+        This storage is SHARED, visible to other apps, and may not be
+        be available if the user didn't put in an external SD card.
+        It will remain untouched when your app is uninstalled.
+
+        Returns None if not found, otherwise path to storage.
+
+        WARNING: You need storage permissions to access this storage.
+                 If it is not writable and presents as empty even with
+                 permissions, then the external sd card may not be present.
+    """
+    Build = autoclass('android.os.Build')
+    if Build.VERSION.SDK_INT >= 24:
```

Comment:
Same here:
```python
VERSION = autoclass('android.os.Build$VERSION')
if VERSION.SDK_INT >= 24:
```
And since we use it twice maybe we want to make it global, same for `Environment`.
What do you think?